src/network/connection.ts

```diff
@@ -291,6 +293,8 @@ export class Connection extends EventEmitter {
       return callback[kCallbackPromise]
     }
 
+    clearInterval(this.#reauthenticationTimeout)
```

Comment:
I think this should be before the previous if, otherwise there might be some odd state machine problems in case of errors.